docs/_static/didman/v1.yaml

```diff
@@ -61,30 +61,67 @@ paths:
         schema:
           type: string
     post:
-      summary: Add an endpoint to a DID Document.
+      operationId: "addEndpoint"
+      summary: Add a service endpoint or a reference to a service.
       description: |
         To distinguish from compound services, this type of service will be called an endpoint.
         In the Nuts specs this type of service is called a concrete service.
 
-        Add an endpoint with a type and URL to a DID Document. The API will convert it to a DID service with the serviceEndpoint set to a URL.
+        Add an endpoint with a type and URL to a DID Service in a Document. The API will convert it to a DID service with the serviceEndpoint set to a URL.
         This API will also check if an endpoint with the same type already exists. This API is not meant to add compound services.
 
+        The URL can either be an Endpoint or a reference to another service.
+
         error returns:
         * 400 - incorrect input
         * 404 - unknown DID
         * 409 - a service with the same type already exists
-      operationId: "addEndpoint"
       tags:
         - DID
       requestBody:
         required: true
         content:
           application/json:
             schema:
-              $ref: '#/components/schemas/EndpointCreateRequest'
+              $ref: '#/components/schemas/EndpointProperties'
       responses:
-        "204":
-          description: The given type and URL have been added as service to the DID Document
+        "200":
+          description: The service which has been added to the DID Document
+          content:
+            application/json:
+              schema:
+                $ref: '#/components/schemas/Endpoint'
+        default:
+          $ref: '../common/error_response.yaml'
+  /internal/didman/v1/did/{did}/endpoint/{type}:
+    parameters:
+      - name: did
+        in: path
+        description: URL encoded DID.
+        required: true
+        example:
+          - "did:nuts:1234"
+        schema:
+          type: string
+      - name: type
+        in: path
+        description: Type of the service
+        required: true
+        example:
+          - "eOverdracht"
+        schema:
+          type: string
+    delete:
+      operationId: deleteEndpointsByType
+      description: Delete all endpoints with the provided type from the DID Document.
+      responses:
+        204:
+          description: |
+            All existing endpoints with with this type were succesfully deleted.
+        400:
```

Comment:
error responses should go in the description. A `default` is added with a schema, the current setup would suggest `400` and `404` do not return a error return body.